### PR TITLE
chore: Run axum server on https

### PR DESCRIPTION
**Pull request by holzeis**
We need to enforce https if we want to support hosting the webapp in the cloud. This change will now host the app solely on https using self-signed certificates. The cert path can be configured through the cli to point to production certs.

### CODE REVIEW

**Review by bonomat — COMMENTED** [review]
Is this the right approach? Wouldn't it make more sense to run it behind an nginx reverse proxy ?

**Comment by holzeis** [comment]
> Is this the right approach? Wouldn't it make more sense to run it behind an nginx reverse proxy ?

Terminating ssl encryption directly on the service is definitely the more secure approach.

I opted against a reverse proxy setup as it would make the deployment more complex (requires an additional service to be run and configured), while adding ssl termination on the service is simpler to setup.

File: webapp/src/cli.rs [hunk 1]
```diff
@@ -74,6 +78,15 @@ impl Opts {
         Ok(data_dir)
     }
 
+    pub fn cert_dir(&self) -> Result<PathBuf> {
+        let cert_dir = match self.cert_dir.clone() {
+            None => current_dir()?.join("webapp/certs"),
```
Comment (luckysori) [inline webapp/src/cli.rs:83]: So if we don't specify and this fallback directory doesn't exist, we would just fail, right?

I guess this fallback is convenient if we run the `webapp` from the root of this repo.

**Review by luckysori — APPROVED** [review]
LGTM.

File: webapp/README.md [hunk 2]
```diff
@@ -16,9 +16,9 @@ flutter build web
 ## Run the rust app
 
 ```bash
-cargo run
+cargo run - --cert-dir certs --data-dir ../data
```
Comment (bonomat) [inline webapp/README.md:19]:
```suggestion
cargo run -- --cert-dir certs --data-dir ../data
```

File: webapp/src/cli.rs (hunk 1 above)
Comment (holzeis) [reply webapp/src/cli.rs:83]: Exactly! 👍

File: webapp/frontend/lib/wallet/wallet_service.dart [hunk 3]
```diff
@@ -15,7 +15,7 @@ class WalletService {
     const host = "localhost";
 
     try {
-      final response = await http.get(Uri.http('$host:$port', '/api/balance'));
+      final response = await http.get(Uri.https('$host:$port', '/api/balance'));
```
Comment (bonomat) [inline webapp/frontend/lib/wallet/wallet_service.dart:18]: After rebase this change should not be necessary

File: webapp/src/main.rs [hunk 4]
```diff
@@ -82,12 +86,23 @@ async fn main() -> Result<()> {
...
+    // configure certificate and private key used by https
+    let config =
+        RustlsConfig::from_pem_file(cert_dir.join("cert.pem"), cert_dir.join("key.pem")).await?;
+
+    let app = router(Arc::new(rx), network);
+
+    // run https server
+    let addr = SocketAddr::from(([0, 0, 0, 0], 3001));
+    tracing::debug!("listening on {}", addr);
+    axum_server::bind_rustls(addr, config)
```
Comment (bonomat) [inline webapp/src/main.rs:98]: I'm still divided whether we should simply run on http and add https using nginx

**Review by bonomat — APPROVED** [review]
LGTM

**Comment by holzeis** [comment]
closed in favor of https://github.com/get10101/10101/pull/1876

**Pull request closed without merging**